cypress/e2e/settings.cy.js

```diff
@@ -1,32 +1,97 @@
 /// <reference types="cypress" />
 /// <reference types="../support" />
 
+import homePageObject from '../support/pages/home.pageObject';
+import settingPageObject from '../support/pages/settings.pageObject';
+import SignInPageObject from '../support/pages/signIn.pageObject'
+
+const faker = require('faker');
+const homePage = new homePageObject();
+const settingsPage = new settingPageObject();
+const signInPage = new SignInPageObject()
+
 describe('Settings page', () => {
-  before(() => {
+  let user;
 
+  const userChanged = {
+    username: faker.name.firstName(),
+    bio: faker.lorem.words(),
+    email: faker.internet.email(),
+    password: faker.internet.password(),
+  };
+
+  before(() => {
+    cy.task('generateUser').then(generateUser => {
+      user = generateUser;
+    });
   });
 
   beforeEach(() => {
-
+    cy.task('db:clear');
+    homePage.visit();
   });
 
   it('should provide an ability to update username', () => {
+    cy.login(user.email, user.username, user.password);
```

Comment:
u use it in every test. try to avoid duplicated code. move it to beforeEach

Reply:
fixed